src/components/NetworkSwitch.tsx

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { upperFirst } from 'lodash'
+import { ArrowRight } from 'lucide-react'
+import posthog from 'posthog-js'
+import { useCallback, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import styled, { useTheme } from 'styled-components'
+
+import Badge from '@/components/Badge'
+import DotIcon from '@/components/DotIcon'
+import Select from '@/components/Inputs/Select'
+import { useAppDispatch, useAppSelector } from '@/hooks/redux'
+import i18next from '@/i18n'
+import ModalPortal from '@/modals/ModalPortal'
+import SettingsModal from '@/modals/SettingsModal'
+import { networkPresetSwitched } from '@/storage/settings/networkActions'
+import { networkPresets } from '@/storage/settings/settingsPersistentStorage'
+import { NetworkName, NetworkNames } from '@/types/network'
+
+interface NetworkSelectOption {
+  label: string
+  value: NetworkName
+}
+
+type NonCustomNetworkName = Exclude<keyof typeof NetworkNames, 'custom' | 'localhost'>
+
+const NetworkSwitch = ({ className }: { className?: string }) => {
+  const { t } = useTranslation()
+  const dispatch = useAppDispatch()
+  const network = useAppSelector((state) => state.network)
+  const [isSettingsModalOpen, setIsSettingsModalOpen] = useState(false)
```

Comment:
nit: I like to leave an empty line between the hooks and the state to easily distinguish the state of the component, but that's a super nit and just a personal preference :)

Reply:
Fixed in https://github.com/alephium/desktop-wallet/pull/614